src/hotspot/cpu/aarch64/codeBuffer_aarch64.cpp

```diff
@@ -26,6 +26,55 @@
 #include "asm/codeBuffer.inline.hpp"
 #include "asm/macroAssembler.hpp"
 
+void CodeBuffer::shared_stub_to_runtime_for(address dest, int caller_offset) {
+  if (_shared_stub_to_runtime_call_requests == nullptr) {
+    _shared_stub_to_runtime_call_requests = new SharedStubToRuntimeCallRequests();
+  }
+  SharedStubToRuntimeCallRequest request(dest, caller_offset);
+  _shared_stub_to_runtime_call_requests->push(request);
+  _finalize_stubs = true;
+}
+
+template <typename MacroAssembler>
+bool emit_shared_stubs_to_runtime_call(CodeBuffer* cb, CodeBuffer::SharedStubToRuntimeCallRequests* requests) {
+  if (requests == NULL) {
+    return true;
+  }
+  auto by_dest = [](CodeBuffer::SharedStubToRuntimeCallRequest* r1, CodeBuffer::SharedStubToRuntimeCallRequest* r2) {
+    if (r1->dest() < r2->dest()) {
+      return -1;
+    } else if (r1->dest() == r2->dest()) {
+      return 0;
+    } else {
+      return 1;
+    }
+  };
+  requests->sort(by_dest);
```

Comment:
While undoubtedly correct, the approach of sorting the requests then deleting duplicates is too heavyweight for this application. A hash table with linear probing (and chaining for calls to the same destination) would be a simple way to solve the problem.